internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -4813,11 +4843,34 @@ func expandKubernetesClusterServiceMeshProfile(input []interface{}, existing *ma
 		}
 
 		profile.Istio.Components.IngressGateways = &istioIngressGatewaysList
+
+		if raw["certificate_authority"] != nil {
+			certificateAuthority := expandKubernetesClusterServiceMeshProfileCertificateAuthority(raw["certificate_authority"].([]interface{}))
+			profile.Istio.CertificateAuthority = certificateAuthority
+		}
 	}
 
 	return &profile
 }
 
+func expandKubernetesClusterServiceMeshProfileCertificateAuthority(input []interface{}) *managedclusters.IstioCertificateAuthority {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+
+	config := input[0].(map[string]interface{})
+
+	return &managedclusters.IstioCertificateAuthority{
+		Plugin: &managedclusters.IstioPluginCertificateAuthority{
+			KeyVaultId:          utils.String(config["key_vault_id"].(string)),
+			RootCertObjectName:  utils.String(config["root_cert_object_name"].(string)),
+			CertChainObjectName: utils.String(config["cert_chain_object_name"].(string)),
+			CertObjectName:      utils.String(config["cert_object_name"].(string)),
+			KeyObjectName:       utils.String(config["key_object_name"].(string)),
```

Comment:
We should use `pointer.To` here 
```suggestion
			KeyVaultId:          pointer.To(config["key_vault_id"].(string)),
			RootCertObjectName:  pointer.To(config["root_cert_object_name"].(string)),
			CertChainObjectName: pointer.To(config["cert_chain_object_name"].(string)),
			CertObjectName:      pointer.To(config["cert_object_name"].(string)),
			KeyObjectName:       pointer.To(config["key_object_name"].(string)),
```